libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -234,12 +234,82 @@ void IRGeneratorForStatements::endVisit(Return const& _return)
 
 void IRGeneratorForStatements::endVisit(UnaryOperation const& _unaryOperation)
 {
-	if (type(_unaryOperation).category() == Type::Category::RationalNumber)
+	auto const& opType = type(_unaryOperation);
+	Token const op = _unaryOperation.getOperator();
+
+	if (opType.category() == Type::Category::RationalNumber)
+	{
 		defineExpression(_unaryOperation) <<
-			formatNumber(type(_unaryOperation).literalValue(nullptr)) <<
+			formatNumber(opType.literalValue(nullptr)) <<
 			"\n";
+	}
+	else if (opType.category() == Type::Category::Integer)
+	{
+		string intermediateValue = m_context.newYulVariable();
+
+		if (op == Token::Inc || op == Token::Dec)
+		{
+			auto const* intType = dynamic_cast<IntegerType const*>(&opType);
+
+			solAssert(!!m_currentLValue, "LValue not retrieved.");
+			string lvalue = m_currentLValue->retrieveValue();
+			m_code << "let " << intermediateValue << " := ";
+
+			if (op == Token::Inc)
+				m_code <<
+					m_utils.overflowCheckedUIntAddFunction(intType->numBits());
+			else
+				m_code << m_utils.overflowCheckedUIntSubFunction();
+
+			m_code << "(" << lvalue << ", 1)\n";
+
+			string previousValue;
+
+			if (!_unaryOperation.isPrefixOperation())
```

Comment:
This code could just do into the `else` branch below.

Reply:
I don't think so. `lvalue` doesn't actually contain the value but only the code to extract the value. If we execute that after `storeValue()` we get the new value

Reply:
I think it would be cleaner to define `previousValue` in both cases (and already use it for the add/sub function call), so that we only have a single `if (isPrefix)`.